### PR TITLE
Update to rust 2015-03-22

### PR DESCRIPTION
**Pull request by tamird**
@alexcrichton

### CODE REVIEW

File: tests/test_cargo_cross_compile.rs [hunk 1]
```diff
@@ -455,6 +455,7 @@ test!(cross_with_a_build_script {
         "#)
         .file("build.rs", format!(r#"
             use std::env;
+            use std::old_path::{{Path, GenericPath}};
```
Comment (daboross) [inline tests/test_cargo_cross_compile.rs]: Is this on purpose double `{{}}` instead of single `{}`?

Comment (tamird) [inline tests/test_cargo_cross_compile.rs]: yes, this is a format string - the escaping is required

Comment (daboross) [inline tests/test_cargo_cross_compile.rs]: Ahhh, ok, that makes sense. Thanks!

**Comment by tamird** [comment]
This travis failure looks like a flake. This should be good to go. r? @alexcrichton

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Comment by tamird** [comment]
@alexcrichton

macs flaked?
http://buildbot.rust-lang.org/builders/cargo-mac-32/builds/1022/steps/test/logs/stdio
http://buildbot.rust-lang.org/builders/cargo-mac-64/builds/1019/steps/test/logs/stdio

windows has an issue with access denied? (in addition to flakes)
http://buildbot.rust-lang.org/builders/cargo-win-32/builds/902/steps/test/logs/stdio
http://buildbot.rust-lang.org/builders/cargo-win-64/builds/546/steps/test/logs/stdio

**Comment by alexcrichton** [comment]
Hm those do not look flaky, I haven't seen many of those before, I think that those failures warrant some investigation.

**Comment by alexcrichton** [comment]
I had [to apply this patch](https://gist.github.com/alexcrichton/1a8d253453a07749c9aa) to get it to pass tests locally.

**Comment by tamird** [comment]
@alexcrichton nice, thanks for tracking that down!

**Comment by alexcrichton** [comment]
@bors: retry

Now *that's* a spurious failure

**Comment by tamird** [comment]
that linux travis build really had some trouble O.o

**Pull request merged**